runtime/gc_include/j9mm.hdf

```diff
@@ -60,5 +60,19 @@ struct J9VMThread;
 		<struct>MM_InterruptCompilationEvent</struct>
 		<data type="struct J9VMThread*" name="currentThread" description="current thread" />
 	</event>
+	
+	<event>
+		<name>J9HOOK_MM_OBJECT_ALLOCATION_SAMPLING</name>
+		<description>
+			Triggered when the object allocations sampling threshold is reached
+		</description>
+		<struct>MM_ObjectAllocationSamplingEvent</struct>
+		<data type="struct J9VMThread*" name="currentThread" description="current thread" />
+		<data type="U_64" name="timestamp" description="time of event" />
+		<data type="UDATA" name="eventid" description="unique identifier for event" />
+		<data type="omrobjectptr_t" name="object" description="the object which has been allocated." />
```

Comment:
Should `omrobjectptr_t` be `j9object_t`?

Reply:
no

Reply:
From following defines:
https://github.com/eclipse/openj9/blob/581167b2533cd545d22f9655e4c2e2eff4b95b8f/runtime/gc_glue_java/objectdescription.h#L32
and
https://github.com/eclipse/openj9/blob/581167b2533cd545d22f9655e4c2e2eff4b95b8f/runtime/oti/j9accessbarrier.h#L62
`omrobjectptr_t` seems equivalent to `j9object_t` from functionality perspective, maybe there is a convention to prefer `omr` version over `j9` one? 

By the way, `JDK8` sanity test failed due to infrastructure issue.
```
java.io.IOException: Tried to load head FlowNodes for execution Owner[PullRequest-Sanity-JDK8-linux_ppc-64_cmprssptrs_le-OpenJ9/825:PullRequest-Sanity-JDK8-linux_ppc-64_cmprssptrs_le-OpenJ9 #825] but FlowNode was not found in storage for head id:FlowNodeId 1:33
```

Reply:
@charliegracie Can you expand on your "no"?  Why is `omrobjectptr_t` the right choice in this code?  Consumers of the hook would be more likely to expect a j9object_t, correct?

Reply:
omrobjectptr_t is technical the right answer.  I had hoped at some point to move this hook into OMR so that other runtimes could use it.  At that point it would have to be `omrobjectptr_t` and not `j9object_t` or `J9Object *`. `j9object_t` == `J9Object *` == `omrobjectptr_t`. 

If it is easier for developers to have it as j9object_t I will change it.

Reply:
The (only) caller of this GC hook is `traceAllocateObject(... J9Object * object, ...)` in this PR, a `J9Object * object` is passed into `TRIGGER_J9HOOK_MM_OBJECT_ALLOCATION_SAMPLING`. 
Since `j9object_t == omrobjectptr_t`, there is no usage difference here. It appears more like a preference of an `OMR` or `J9` kind of definition in current code space.